### PR TITLE
feat: Add delegate_collection_authority and revoke_collection_authority

### LINKED ISSUE #16: Add `delegate_collection_authority`
In our Node.js SDK, we have the [`delegateCollectionAuthority` method](https://github.com/helius-labs/helius-sdk/blob/2a50e1ce6b33730134e2422723aa7c98265fbcd5/src/Helius.ts#L409-L467), which delegates a collection authority to a new address. This is handy when used with our Mint API. We should add this functionality to the Rust SDK

### PR DESCRIPTION
**Pull request by grandima**
### Closes #16  #17 
This PR:

- adds possibility to delegate and revoke collection authority to Helius SDK
- adds example usage on how to delegate and revoke collection authority
- changes `MintApiAuthority` associated value from `&'static str` to `Pubkey`

### CODE REVIEW

**Comment by 0xIchigo** [comment]
This is extremely based @grandima ! I'm just wondering: do you have any real-world examples of delegating and revoking the collection authority, or is it just limited to the tests? I'll give this PR a more formal review soon — I am currently tied up with other dev work that's taking priority. Nonetheless, thank you for your contributions! 🔥

**Comment by grandima** [comment]
Thanks for acknowledgment!

No, I have not seen real-world examples. But I've included [one](https://github.com/helius-labs/helius-rust-sdk/pull/77/files#diff-b33eb4515516bb2d6bf5b8285ee1a68d76e6a1be4796bfce45a5589c9feb0258) in this PR.

File: tests/test_mint_api.rs [hunk 1]
```diff
@@ -160,44 +160,67 @@ async fn test_get_asset_proof_failure() {
...
-    let devnet_cluster: Cluster = Cluster::Devnet;
-    let mainnet_cluster: Cluster = Cluster::MainnetBeta;
-
-    let devnet_authority: std::result::Result<MintApiAuthority, &str> = MintApiAuthority::from_cluster(devnet_cluster);
-    let mainnet_authority: std::result::Result<MintApiAuthority, &str> =
-        MintApiAuthority::from_cluster(mainnet_cluster);
-
-    assert_ne!(
-        devnet_authority.unwrap(),
-        MintApiAuthority::Devnet("Blade"),
```
Comment (0xIchigo) [inline tests/test_mint_api.rs:195]: Only fitting that this is merged on the same day the new Blade movie is delayed indefinitely

**Review by 0xIchigo — APPROVED** [review]
This is extremely based! Thank you for taking the time to implement this! Everything looks good on my end 🔥

**Pull request merged**